TAS_Engine/Compute/TBDSizing.cs

```diff
@@ -0,0 +1,120 @@
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Base;
+using BH.oM.Geometry;
+
+using TAS3D;
+using TBD;
+using TSD;
+
+using System.ComponentModel;
+using BH.oM.Reflection.Attributes;
+using System.Collections.Generic;
+using System.Linq;
+
+using System.Runtime.InteropServices;
+using BH.oM.Environment.Elements;
+
+using BH.oM.Environment.Fragments;
+using System;
+
+namespace BH.Engine.TAS
+{
+    public static partial class Compute
+    {
+
+        [Description("BH.Engine.TAS Compute - Run Design Day Sizing")]
+        [Input("tbdFile", "The full file path of the TBD file")]
+        [Input("run", "Set to true when you want to run the component, default false")]
+        [Output("success", "True if sizing has been successfully run, false otherwise")]
+        public static bool TBDDesignDaySizing(string tbdFile, bool run = false)
+        {
+
+            if (!run) return false;
+
+            if (string.IsNullOrEmpty(tbdFile))
+            {
+                BH.Engine.Reflection.Compute.RecordError("Please provide a valid TBD file path");
+                return false;
+            }
+
+            TBDDocument tbdDocument = new TBDDocument();
+            tbdDocument.open(tbdFile);
+            if (tbdDocument != null)
+                tbdDocument.sizing(0);
+            tbdDocument.save();
+            tbdDocument.close();
+
+            int intrefcountTBD = 0;
+            do
+            {
+                intrefcountTBD = Marshal.FinalReleaseComObject(tbdDocument);
+            } while (intrefcountTBD > 0);
+            tbdDocument = null;
+
+            return true;
+
+        }
+
+        [Description("BH.Engine.TAS Compute - Run Dynamic Simulation")]
+        [Input("tbdFile", "The full file path of the TBD file")]
+        [Input("startDay", "Set start day, default 0")]
+        [Input("endDay", "Set end day, default 365")]
+        [Input("autoViewResults", "Set to true when you want to open results when simulation complete, default false")]
+        [Input("run", "Set to true when you want to run the component, default false")]
+        [Output("success", "True if simulation has been successfully run, false otherwise")]
+        public static bool TBDDynamicSimulation(string tbdFile, int startDay = 0, int endDay = 365, bool autoViewResults = false, bool run = false)
+        {
+
+            if (!run) return false;
+
+            if (!run) return false;
+
+            if (string.IsNullOrEmpty(tbdFile))
+            {
+                BH.Engine.Reflection.Compute.RecordError("Please provide a valid TBD file path");
+                return false;
+            }
+
+            TBDDocument tbdDocument = new TBDDocument();
+            tbdDocument.open(tbdFile);
+
+            string tsdFile = tbdFile.Replace(".tbd", ".tsd");
+
+            tbdDocument.simulate(startDay, endDay, autoViewResults ? 1 : 0, 1, 0, 0, tsdFile, 0, 0);
```

Comment:
Check the `tbdDocument` is not equal to `null` first?

Reply:
done